packages/visualizations/addon/models/table.js

```diff
@@ -92,12 +92,25 @@ export default VisualizationBase.extend(Validations, {
       };
     });
 
+    let newColumns = [
```

Comment:
I feel `rebuildConfig` here has grown to be a bit too big. Can we split this into multiple smaller methods?
`getMetricColumns`, `getDimensionColumns`...